Source/RapyutaSimulationPlugins/Private/RRROS2GameMode.cpp

```diff
@@ -10,9 +13,39 @@
 #include "Tools/RRROS2ClockPublisher.h"
 #include "Tools/SimulationState.h"
 
-void ARRROS2GameMode::BeginPlay()
+void ARRROS2GameMode::InitGame(const FString& InMapName, const FString& InOptions, FString& OutErrorMessage)
+{
+    Super::InitGame(InMapName, InOptions, OutErrorMessage);
+    UE_LOG(LogRapyutaCore,
+           Log,
+           TEXT("[ARRROS2GameMode] INIT GAME [%s/%s] - Options: %s\n%s"),
+           *InMapName,
+           *GetWorld()->GetName(),
+           *InOptions,
+           *OutErrorMessage);
+    UE_LOG(LogRapyutaCore,
+           Log,
+           TEXT("NUM OF CPU CORES: [%d] - WITH HYPERTHREADS: [%d] - RECOMMENDED NUM OF WORKER THREADS: [%d]"),
+           FPlatformMisc::NumberOfCores(),
+           FPlatformMisc::NumberOfCoresIncludingHyperthreads(),
+           FPlatformMisc::NumberOfWorkerThreadsToSpawn());
+    UE_LOG(LogRapyutaCore, Display, TEXT("ShouldUseThreadingForPerformance: %d"), FApp::ShouldUseThreadingForPerformance());
+
+    // Init Sim main components
+    InitSim();
+}
+
+void ARRROS2GameMode::InitSim()
```

Comment:
what is the intension of this method?

Reply:
Ah right, I thought you meant [`InitGame()`](https://github.com/EpicGames/UnrealEngine/blob/release/Engine/Source/Runtime/Engine/Classes/GameFramework/GameModeBase.h#L61).
This one `InitSim()` as made `virtual` is mostly for child game modes to override and add their possible custom Sim-global component initializing.